### PR TITLE
Export the Tee type via Data.Fold

### PR DESCRIPTION
**Pull request by rnjtranjan**

### CODE REVIEW

**Comment by rnjtranjan** [comment]
Fix:https://github.com/composewell/streamly/issues/2154

File: core/src/Streamly/Data/Fold/Tee.hs [hunk 1]
```diff
@@ -1,3 +1,4 @@
+{-# OPTIONS_GHC -Wno-deprecations #-}
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Fold/Tee.hs:1]: This should not be needed.

File: core/src/Streamly/Data/Fold/Tee.hs [hunk 2]
```diff
@@ -6,34 +7,8 @@
...
--- input to both the folds and combine the outputs using Monoid or Semigroup
--- instances of the output types:
---
--- >>> import Data.Monoid (Sum(..))
--- >>> t = Tee Fold.one <> Tee Fold.latest
--- >>> Stream.fold (toFold t) (fmap Sum $ Stream.enumerateFromTo 1.0 100.0)
--- Just (Sum {getSum = 101.0})
---
--- The 'Num', 'Floating', and 'Fractional' instances work in the same way.
---
```
Comment (harendra-kumar) [inline core/src/Streamly/Data/Fold/Tee.hs:35]: We should note remove this.

File: core/src/Streamly/Internal/Data/Fold/Tee.hs [hunk 3]
```diff
@@ -10,6 +10,7 @@
 -- 'Semigroup', 'Monoid', 'Num', 'Floating' and 'Fractional' instances.
 --
 module Streamly.Internal.Data.Fold.Tee
+{-# DEPRECATED "Please use \"Streamly.Internal.Data.Fold\" instead." #-}
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Fold/Tee.hs:13]: Do not deprecate this. Just import this module in Streamly.Data.Fold.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

File: core/src/Streamly/Internal/Data/Fold/Tee.hs [hunk 4]
```diff
@@ -9,6 +9,31 @@
...
+-- input to both the folds and combine the outputs using Monoid or Semigroup
+-- instances of the output types:
+--
+-- >>> import Data.Monoid (Sum(..))
+-- >>> t = Tee Fold.one <> Tee Fold.latest
+-- >>> Stream.fold (unTee t) (fmap Sum $ Stream.enumerateFromTo 1.0 100.0)
+-- Just (Sum {getSum = 101.0})
+--
+-- The 'Num', 'Floating', and 'Fractional' instances work in the same way.
+--
```
Comment (harendra-kumar) [inline core/src/Streamly/Internal/Data/Fold/Tee.hs:36]: Move the documentation to the Tee type so that it is visible to users.

**Review by harendra-kumar — CHANGES_REQUESTED** [review]

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**